script.js

```diff
@@ -1,7 +1,533 @@
+  li.innerText = i;
+  tech1.appendChild(li);
+}
+div2.appendChild(tech1);
+div1.appendChild(div2);
+const button1 = document.createElement('button');
+button1.type = 'button';
+button1.className = 'button1';
+button1.innerText = p1.button;
+button1.onclick = function () { myFunction2(); };
+div1.appendChild(button1);
+
+// first card
+const project_desktop = document.getElementById('project_desktop');
+project_desktop.className = 'project_desktop';
+const div3 = document.createElement('div');
+div3.className = 'projects_art';
+
+const h32 = document.createElement('h3');
+h32.innerText = p2.title;
+div3.appendChild(h32);
+project_desktop.appendChild(div3);
+
+const para2 = document.createElement('p');
+para2.innerText = p2.description;
+div3.appendChild(para2);
+
+const div4 = document.createElement('div');
+div4.className = 'sites12';
+const tech2 = document.createElement('ul');
+for (const i of p2.tech) {
+  const li = document.createElement('li');
+  li.innerText = i;
+  tech2.appendChild(li);
+}
+div4.appendChild(tech2);
+div3.appendChild(div4);
+const button2 = document.createElement('button');
+button2.type = 'button';
+button2.className = 'button2';
+button2.innerText = p2.button;
+button2.onclick = function () { myFunction2(); };
+div3.appendChild(button2);
+
+// second card
+const div5 = document.createElement('div');
+div5.className = 'projects_data';
+
+const h33 = document.createElement('h3');
+h33.innerText = p3.title;
+div5.appendChild(h33);
+project_desktop.appendChild(div5);
+
+const para3 = document.createElement('p');
+para3.innerText = p3.description;
+div5.appendChild(para3);
+
+const div6 = document.createElement('div');
+div6.className = 'sites12';
+const tech3 = document.createElement('ul');
+for (const i of p3.tech) {
+  const li = document.createElement('li');
+  li.innerText = i;
+  tech3.appendChild(li);
+}
+div6.appendChild(tech3);
+div5.appendChild(div6);
+const button3 = document.createElement('button');
+button3.type = 'button';
+button3.className = 'button2';
+button3.innerText = p3.button;
+button3.onclick = function () { myFunction2(); };
+div5.appendChild(button3);
+
+// third card
+const div7 = document.createElement('div');
+div7.className = 'projects_web';
+
+const h34 = document.createElement('h3');
+h34.innerText = p4.title;
+div7.appendChild(h34);
+project_desktop.appendChild(div7);
+
+const para4 = document.createElement('p');
+para4.innerText = p4.description;
+div7.appendChild(para4);
+
+const div8 = document.createElement('div');
+div8.className = 'sites12';
+const tech4 = document.createElement('ul');
+for (const i of p4.tech) {
+  const li = document.createElement('li');
+  li.innerText = i;
+  tech4.appendChild(li);
+}
+div8.appendChild(tech4);
+div7.appendChild(div8);
+const button4 = document.createElement('button');
+button4.type = 'button';
+button4.className = 'button2';
+button4.innerText = p4.button;
+button4.onclick = function () { myFunction2(); };
+div7.appendChild(button4);
```

Comment:
- [ ] You are basically assigning each project to HTML structure through DOM one by one. You can still do it in Javascript dynamically. Read more about how you can use iterations of the array of objects.

Reply:
Okay I've use the iterations of the array of objects to repeat the desktop cards.